src/Lean/Widget/InteractiveGoal.lean

```diff
@@ -13,7 +13,10 @@ namespace Lean.Widget
 open Server
 
 structure InteractiveHypothesis where
-  names : Array String
+  /-- The user-friendly name for each hypothesis.
+  If anonymous then the name is inaccessible. -/
+  names : Array Name
+  fvarIds? : Option (Array FVarId)
```

Comment:
When is this `none`?

Reply:
I think it should just be `Array FVarId`. We represent it as optional in the client so that the client can parse data received from older servers, but the server can rely on the client being up-to-date (and anyway extra JSON fields are no issue for JS).

Reply:
yes the none was for backwards compat but you are right it can just give the extra field

Reply:
fixed